src/boot_O2_g3/fault.c

```diff
@@ -37,34 +81,44 @@ const char* sFpuExceptions[] = {
 };
 
 void Fault_SleepImpl(u32 duration) {
-    u64 value = (duration * OS_CPU_COUNTER) / 1000ULL;
+    OSTime value = (duration * OS_CPU_COUNTER) / 1000ULL;
+
     Sleep_Cycles(value);
 }
 
-void Fault_AddClient(FaultClient* client, void* callback, void* param0, void* param1) {
+/**
+ * Registers a fault client.
+ *
+ * Clients contribute at least one page to the crash screen, drawn by `callback`.
+ * Arguments are passed on to the callback through `arg0` and `arg1`.
+ *
+ * The callback is intended to be
+ * `void (*callback)(void* arg0, void* arg1)`
```

Comment:
Probably an @Thar0 question, but any reason not to make a `FaultClientCallback` typedef and use it here and in the struct, instead of `void*`. Same question for the `AddrConvClients`.

Reply:
I guess it is because some clients don't pass a function with the proper signature. For example 
```c

void CIC6105_PrintRomInfo(void) {
    FaultDrawer_DrawText(80, 200, "SP_STATUS %08x", HW_REG(SP_STATUS_REG, u32));
    FaultDrawer_DrawText(40, 184, "ROM_F [Creator:%s]", gBuildTeam);
    FaultDrawer_DrawText(56, 192, "[Date:%s]", gBuildDate);
}

void CIC6105_AddRomInfoFaultPage(void) {
    Fault_AddClient(&romInfoFaultClient, CIC6105_PrintRomInfo, NULL, NULL);
}
```